language/vm/vm-runtime/src/code_cache/script_cache.rs

```diff
@@ -66,24 +79,42 @@ impl<'alloc> ScriptCache<'alloc> {
             }
         };
 
-        match VerifiedScript::new(script) {
-            Ok(script) => Ok(script),
-            Err((_, mut errs)) => {
-                warn!(
-                    "[VM] bytecode verifier returned errors for script: {:?}",
-                    errs
-                );
-                // If there are errors there should be at least one otherwise there's an internal
-                // error in the verifier. We only give back the first error. If the user wants to
-                // debug things, they can do that offline.
-                let error = if errs.is_empty() {
-                    VMStatus::new(StatusCode::VERIFIER_INVARIANT_VIOLATION)
-                } else {
-                    errs.remove(0)
-                };
-
-                Err(error)
+        let mut errs = match VerifiedScript::new(script) {
+            Ok(script) => {
+                // verify dependencies
+                let script_module = script.self_handle();
+                let mut deps = vec![];
+                for module in script.module_handles() {
+                    if module == script_module {
+                        continue;
+                    }
+                    let module_id = ModuleId::new(
+                        *script.as_inner().address_at(module.address),
+                        script.as_inner().identifier_at(module.name).to_owned(),
+                    );
+                    deps.push(load_and_verify_module_id(&module_id, context)?);
+                }
+                let errs = verify_script_dependencies(&script, &deps);
```

Comment:
this is obviously not how we will do it in a very near future but it should be good enough for now

Reply:
don't think so, again the way we are going to rework the loader this will have to go, there is no way to keep this around so I don't feel a TODO would add much.
Sounds good?